auth/src/main/java/com/vimeo/networking2/Authenticator.kt

```diff
@@ -158,6 +183,53 @@ interface Authenticator {
         callback: VimeoCallback<VimeoAccount>
     ): VimeoRequest
 
+    /**
+     * Obtain the URI which can be used to log in the user and receive back a URI that can be exchanged using
+     * [authenticateWithCodeGrant] for a logged in account.
+     *
+     * @param responseCode The response code that can be used to identify the origin of the redirect URI.
+     *
+     * @return The URI which can be opened in a browser.
+     */
+    fun obtainCodeGrantAuthorizationUri(responseCode: Int): String
```

Comment:
Just curious but why isn't this method or the related ones using `VimeoRequest` and `VimeoCallback` ?

Reply:
Well this function is just constructing a URL so it doesn't enqueue a request or require a callback.

Reply:
Ah, I saw the use of `authService` in `AuthenticatorImpl` and assumed it was a call, didn't notice the `request().url().toString()`, makes sense now 👍

Reply:
FWIW, Nick's confusion around this was shared. Most of what the Authenticator does is make requests to our servers. This method is the exception. When I first saw it in action in the example, I wasn't immediately sure if the source of the string was from the server or not or why it was different. Once I looked at the implementation, I understood what was happening, but I was initially a little confused. That's why I made the thread suggestion initially. If you have any other ideas of docs that could call this out in some way, it would be helpful. Completely optional though.